### PR TITLE
Hotfix/additional form options error case

### PR DESCRIPTION
**Pull request by carnage**
Fixes an additional case of the previous bug where composed objects overwrite options specified in other annotations

### CODE REVIEW

**Comment by jaapio** [comment]
please add a new testcase for this fix, rather then changing an existing one.

File: tests/ZendTest/Form/Annotation/AnnotationBuilderTest.php [hunk 1]
```diff
@@ -198,7 +199,7 @@ public function testOptionsAnnotationAndComposedObjectAnnotation($childName)
 
         $child = $form->get($childName);
 
-        $target = $child->getTargetElement();
+        $target = ($child instanceof Collection)? $child->getTargetElement() : $child;
```
Comment (keradus) [inline tests/ZendTest/Form/Annotation/AnnotationBuilderTest.php]: I would add a space before that `?`

**Comment by carnage** [comment]
@jappio This fix is for another occurance of the bug that the test is for, it extends the test case via a data provider, so in this instance I don't think a separate test case is warrented.

**Comment by jaapio** [comment]
Changing a test is always a bad thing. The test might have the same intension as the issue your are testing. But somehow you found a bug in the behavior, so you need te be sure you don't break anything else. Leaving existing tests as they are will help you checking this.

**Comment by Ocramius** [comment]
Agree with @jaapio here: can a new test be written instead?

**Comment by carnage** [comment]
ok I'll c+p the test into a new method.

Linking #7108 for reference

**Comment by carnage** [comment]
Build failures seem unrelated. Is this good to merge now @weierophinney

**Comment by carnage** [comment]
reopened against develop #7309

**Pull request closed without merging**